src/Features/Core/Portable/UseConditionalExpression/ForAssignment/AbstractUseConditionalExpressionForAssignmentCodeFixProvider.cs

```diff
@@ -0,0 +1,238 @@
+
+            if (!UseConditionalExpressionForAssignmentHelpers.TryMatchPattern(
+                    syntaxFacts, ifOperation,
+                    out var trueAssignment, out var falseAssignment))
+            {
+                return;
+            }
+
+            var conditionalExpression = await CreateConditionalExpressionAsync(
+                document, ifOperation, trueAssignment.Parent, falseAssignment.Parent,
+                trueAssignment.Value, falseAssignment.Value, 
+                trueAssignment.IsRef, cancellationToken).ConfigureAwait(false);
+
+            // See if we're assigning to a variable declared directly above the if statement. If so,
+            // try to inline the conditional directly into the initializer for that variable.
+            if (!TryConvertWhenAssignmentToLocalDeclaredImmediateAbove(
+                    syntaxFacts, editor, ifOperation, 
+                    trueAssignment, falseAssignment, conditionalExpression))
+            {
+                // If not, just replace the if-statement with a single assignment of the new
+                // conditional.
+                ConvertOnlyIfToConditionalExpression(
+                    editor, ifOperation, trueAssignment, falseAssignment, conditionalExpression);
+            }
+        }
+
+        private void ConvertOnlyIfToConditionalExpression(
+            SyntaxEditor editor, IConditionalOperation ifOperation,
+            ISimpleAssignmentOperation trueAssignment, ISimpleAssignmentOperation falseAssignment,
+            TExpressionSyntax conditionalExpression)
+        {
+            var generator = editor.Generator;
+            var ifStatement = (TIfStatementSyntax)ifOperation.Syntax;
+            var expressionStatement = (TStatementSyntax)generator.ExpressionStatement(
+                generator.AssignmentStatement(
+                    trueAssignment.Target.Syntax,
+                    conditionalExpression)).WithTriviaFrom(ifStatement);
+
+            editor.ReplaceNode(
+                ifOperation.Syntax,
+                this.WrapWithBlockIfAppropriate(ifStatement, expressionStatement));
+        }
+
+        private bool TryConvertWhenAssignmentToLocalDeclaredImmediateAbove(
+            ISyntaxFactsService syntaxFacts, SyntaxEditor editor, IConditionalOperation ifOperation,
+            ISimpleAssignmentOperation trueAssignment, ISimpleAssignmentOperation falseAssignment,
+            TExpressionSyntax conditionalExpression)
+        {
+            if (!TryFindMatchingLocalDeclarationImmediatelyAbove(
+                    ifOperation, trueAssignment, falseAssignment,
+                    out var localDeclarationOperation))
+            {
+                return false;
+            }
+
+            // We found a valid local declaration right above the if-statement.
+            var localDeclaration = localDeclarationOperation.Syntax;
+            var declarator = localDeclarationOperation.Declarations[0].Declarators[0];
```

Comment:
>var declarator = localDeclarationOperation.Declarations[0].Declarators[0]; [](start = 12, length = 74)

what if there are multiple declarations just above the if?
e.g.

int x = 5, y = M();
if (condition)
{
 x = 4;
}
else
{
 x = 6;
}

Reply:
Thank you! I see.
I just want to mention that this can be found when reviewing the method below but not the current one.

I'd add a comment saying something like: 
// The call of TryFindMatchingLocalDeclarationImmediatelyAbove above guarantees a single Declaration and a single Declarator.


---
In reply to: [185656240](https://github.com/dotnet/roslyn/pull/26572#discussion_r185656240) [](ancestors = 185656240)

Reply:
I updated the helper method to pass back the declarator as well.  that just keeps things nice and simple.